iroh-relay/src/main.rs

```diff
@@ -215,10 +225,62 @@ impl From<AccessConfig> for iroh_relay::server::AccessConfig {
                     .boxed()
                 }))
             }
+            AccessConfig::Http(url) => {
+                let client = reqwest::Client::default();
+                iroh_relay::server::AccessConfig::Restricted(Box::new(move |node_id| {
+                    let client = client.clone();
+                    let url = url.clone();
+                    async move { http_access_check(&client, url, node_id).await }.boxed()
+                }))
+            }
         }
     }
 }
 
+#[tracing::instrument("http-access-check", skip_all, fields(node_id=%node_id.fmt_short()))]
+async fn http_access_check(
+    client: &reqwest::Client,
+    url: Url,
+    node_id: NodeId,
+) -> iroh_relay::server::Access {
+    use iroh_relay::server::Access;
+    debug!(%url, "Check relay access via HTTP POST");
+    let res = match client
+        .post(url)
+        .header("X-Iroh-NodeId", node_id.to_string())
+        .send()
+        .await
+    {
+        Ok(t) => t,
```

Comment:
You could use the fany(ish) new(ish) let-else expression:

`let Ok(res) = client.post...await else { warn!(...); return Access::Deny; }`?

Reply:
With `let-else` you can't print the error in the warn message, which I think is helpful, no?